src/daft-sql/src/modules/url.rs

```diff
@@ -0,0 +1,192 @@
+use std::sync::Arc;
+
+use daft_dsl::{Expr, ExprRef, LiteralValue};
+use daft_functions::uri::{download, download::DownloadFunction, upload, upload::UploadFunction};
+use sqlparser::ast::FunctionArg;
+
+use super::SQLModule;
+use crate::{
+    error::{PlannerError, SQLPlannerResult},
+    functions::{SQLFunction, SQLFunctionArguments, SQLFunctions},
+    modules::config::expr_to_iocfg,
+    planner::SQLPlanner,
+    unsupported_sql_err,
+};
+
+pub struct SQLModuleURL;
+
+impl SQLModule for SQLModuleURL {
+    fn register(parent: &mut SQLFunctions) {
+        parent.add_fn("url_download", UrlDownload);
+        parent.add_fn("url_upload", UrlUpload);
+    }
+}
+
+impl TryFrom<SQLFunctionArguments> for DownloadFunction {
+    type Error = PlannerError;
+
+    fn try_from(args: SQLFunctionArguments) -> Result<Self, Self::Error> {
+        let max_connections = args.try_get_named("max_connections")?.unwrap_or(32);
```

Comment:
let’s not have magic numbers. maybe give this a const value and perhaps include in python documentation,

Reply:
Use `const DEFAULT_MAX_CONNECTIONS: usize = 32` instead.
![image](https://github.com/user-attachments/assets/6ef75403-7fd9-4d8f-ac24-95212f7b519c)

Reply:
lgtm